hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/placement/algorithms/SCMContainerPlacementRackAware.java

```diff
@@ -124,11 +126,11 @@ protected List<DatanodeDetails> chooseDatanodesInternal(
     int datanodeCount = networkTopology.getNumOfLeafNode(NetConstants.ROOT);
     int excludedNodesCount = excludedNodes == null ? 0 : excludedNodes.size();
     int usedNodesCount = usedNodes == null ? 0 : usedNodes.size();
-    if (datanodeCount < nodesRequired + excludedNodesCount + usedNodesCount) {
-      throw new SCMException("No enough datanodes to choose. " +
-          "TotalNode = " + datanodeCount + " RequiredNode = " + nodesRequired +
-          " ExcludedNode = " + excludedNodesCount +
-          " UsedNode = " + usedNodesCount, null);
+    Set<DatanodeDetails> unavailableNodes = JavaUtils.unionOfCollections(usedNodes, excludedNodes);
```

Comment:
unavailableNodes is not used, just its count. so also can get unavailableCount = usedNodesCount + excludedNodesCount
currently, unionOfCollection seems un-necessary for this case